src/main/java/com/waverleysoftware/pageobjects/BasePage.java

```diff
@@ -0,0 +1,56 @@
+package com.waverleysoftware.pageobjects;
+
+import org.openqa.selenium.By;
+import org.openqa.selenium.Keys;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.ui.WebDriverWait;
+
+import java.util.List;
+
+import static org.openqa.selenium.support.ui.ExpectedConditions.stalenessOf;
+import static org.openqa.selenium.support.ui.ExpectedConditions.visibilityOfElementLocated;
+
+/**
+ * A simple base class for processing common methods for all children classes.
+ */
+public class BasePage {
+
+    protected final WebDriver driver;
+    protected final WebDriverWait wait;
+
+    public BasePage(final WebDriver driver) {
+        this.wait = new WebDriverWait(driver, 10);
+        this.driver = driver;
+    }
+
+    public void click(final By locator) {
+        find(locator).click();
+    }
+
+    public void type(final By locator, final String text) {
+        click(locator);
+        if (text != null) {
+            final String existingName = find(locator).getAttribute("value");
+            if (!text.equals(existingName)) {
+                find(locator).clear();
+                find(locator).sendKeys(text);
+            }
+        }
+    }
+
+    public List<WebElement> findAll(final By locator) {
```

Comment:
do you really want to expose `find` method outside?.. we create wrappers to make life easier... but in case of exposed `find` methods, you allows users misusing your API... it makes no sense to call `find`, when you already have wrappers for `click`, `type`, etc.